README.rst

```diff
@@ -0,0 +1,415 @@
+
+    # Define dataset...
+    train_dataset = ...
+
+    # Partition dataset among workers using DistributedSampler
+    train_sampler = torch.utils.data.distributed.DistributedSampler(
+        train_dataset, num_replicas=hvd.size(), rank=hvd.rank())
+
+    train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=..., sampler=train_sampler)
+
+    # Build model...
+    model = ...
+    model.cuda()
+
+    optimizer = optim.SGD(model.parameters())
+
+    # Add Horovod Distributed Optimizer
+    optimizer = hvd.DistributedOptimizer(optimizer, named_parameters=model.named_parameters())
+
+    # Broadcast parameters from rank 0 to all other processes.
+    hvd.broadcast_parameters(model.state_dict(), root_rank=0)
+
+    for epoch in range(100):
+       for batch_idx, (data, target) in enumerate(train_loader):
+           optimizer.zero_grad()
+           output = model(data)
+           loss = F.nll_loss(output, target)
+           loss.backward()
+           optimizer.step()
+           if batch_idx % args.log_interval == 0:
+               print('Train Epoch: {} [{}/{}]\tLoss: {}'.format(
+                   epoch, batch_idx * len(data), len(train_sampler), loss.item()))
+
+
+**Note**: PyTorch support requires NCCL 2.2 or later. It also works with NCCL 2.1.15 if you are not using RoCE or InfiniBand.
+
+mpi4py
+------
+Horovod supports mixing and matching Horovod collectives with other MPI libraries, such as `mpi4py <https://mpi4py.scipy.org>`_,
+provided that the MPI was built with multi-threading support.
+
+You can check for MPI multi-threading support by querying the ``hvd.mpi_threads_supported()`` function.
+
+.. code-block:: bash
+
+    import horovod.tensorflow as hvd
+
+    # Initialize Horovod
+    hvd.init()
+
+    # Verify that MPI multi-threading is supported.
+    assert hvd.mpi_threads_supported()
+
+    from mpi4py import MPI
+    assert hvd.size() == MPI.COMM_WORLD.Get_size()
+
+
+Inference
+---------
+Learn how to optimize your model for inference and remove Horovod operations from the graph `here <docs/inference.md>`_.
+
+
+Tensor Fusion
+-------------
+One of the unique things about Horovod is its ability to interleave communication and computation coupled with the ability
+to batch small *allreduce* operations, which results in improved performance. We call this batching feature Tensor Fusion.
+
+See `here <docs/tensor-fusion.md>`_ for full details and tweaking instructions.
+
+
+Analyzing Horovod Performance
+-----------------------------
+Horovod has the ability to record the timeline of its activity, called Horovod Timeline.
+
+.. image:: https://user-images.githubusercontent.com/16640218/29735271-9e148da0-89ac-11e7-9ae0-11d7a099ac89.png
+   :alt: Horovod Timeline
+
+See `here <docs/timeline.md>`_ for full details and usage instructions.
+
+
+Guides
+------
+1. Run distributed training in Microsoft Azure using `Batch AI and Horovod <https://github.com/Azure/BatchAI/tree/master/recipes/Horovod>`_.
```

Comment:
Remove the "1." for now

When there is more than one link, it can be an unordered list